src/Microsoft.ML.Data/DataLoadSave/Binary/Codecs.cs

```diff
@@ -657,74 +655,62 @@ public Reader(DateTimeCodec codec, Stream stream, int items)
             }
 
-            public override IValueWriter<DvDateTimeZone> OpenWriter(Stream stream)
+            public override IValueWriter<DateTimeOffset> OpenWriter(Stream stream)
             {
                 return new Writer(this, stream);
             }
 
-            public override IValueReader<DvDateTimeZone> OpenReader(Stream stream, int items)
+            public override IValueReader<DateTimeOffset> OpenReader(Stream stream, int items)
             {
                 return new Reader(this, stream, items);
             }
 
-            private sealed class Writer : ValueWriterBase<DvDateTimeZone>
+            private sealed class Writer : ValueWriterBase<DateTimeOffset>
             {
                 private List<short> _offsets;
                 private List<long> _ticks;
 
-                public Writer(DateTimeZoneCodec codec, Stream stream)
+                public Writer(DateTimeOffsetCodec codec, Stream stream)
                     : base(codec.Factory, stream)
                 {
                     _offsets = new List<short>();
                     _ticks = new List<long>();
                 }
 
-                public override void Write(ref DvDateTimeZone value)
+                public override void Write(ref DateTimeOffset value)
                 {
                     Contracts.Assert(_offsets != null, "writer was already committed");
 
-                    var ticks = value.ClockDateTime.Ticks;
-                    var offset = value.OffsetMinutes;
+                    _ticks.Add(value.DateTime.Ticks);
 
-                    _ticks.Add(ticks.RawValue);
-                    if (ticks.IsNA)
-                    {
-                        Contracts.Assert(offset.IsNA);
-                        _offsets.Add(0);
-                    }
-                    else
-                    {
-                        Contracts.Assert(
-                            offset.RawValue >= DvDateTimeZone.MinMinutesOffset &&
-                            offset.RawValue <= DvDateTimeZone.MaxMinutesOffset);
-                        Contracts.Assert(0 <= ticks.RawValue && ticks.RawValue <= DvDateTime.MaxTicks);
-                        _offsets.Add(offset.RawValue);
-                    }
+                    //DateTimeOffset exposes its offset as a TimeSpan, but internally it uses short and in minutes.
+                    //https://github.com/dotnet/coreclr/blob/9499b08eefd895158c3f3c7834e185a73619128d/src/System.Private.CoreLib/shared/System/DateTimeOffset.cs#L51-L53
+                    //https://github.com/dotnet/coreclr/blob/9499b08eefd895158c3f3c7834e185a73619128d/src/System.Private.CoreLib/shared/System/DateTimeOffset.cs#L286-L292
+                    //From everything online(ISO8601, RFC3339, SQL Server doc, the offset supports the range -14 to 14 hours, and only supports minute precision.
+                    _offsets.Add((short)(value.Offset.Ticks / TimeSpan.TicksPerMinute));
```

Comment:
Why can't this just be `value.TotalMinutes`?

Reply:
don't see totalMinutes under value, did you mean under offset?

---
In reply to: [213152272](https://github.com/dotnet/machinelearning/pull/693#discussion_r213152272) [](ancestors = 213152272)

Reply:
Yes, value.Offset.TotalMinutes